lib/src/lifecycle_module.dart

```diff
@@ -83,23 +107,45 @@ abstract class LifecycleModule {
         new StreamController<LifecycleModule>.broadcast();
     _didUnloadChildModuleController =
         new StreamController<LifecycleModule>.broadcast();
+    _willSuspendController = new StreamController<LifecycleModule>.broadcast();
+    _didSuspendController = new StreamController<LifecycleModule>.broadcast();
+    _willResumeController = new StreamController<LifecycleModule>.broadcast();
+    _didResumeController = new StreamController<LifecycleModule>.broadcast();
+
+    _logger = new Logger('w_module');
   }
 
+  /// Whether the module is currently loaded.
+  bool get isLoaded => _isLoaded;
+
+  /// Whether the module is currently suspended.
+  ///
+  /// This will always be false when the module is not loaded.
+  bool get isSuspended => _isSuspended;
+
   //--------------------------------------------------------
   // Public methods that can be used directly to trigger
   // module lifecycle / check current lifecycle state
   //--------------------------------------------------------
 
   /// Public method to trigger the loading of a Module.
+  ///
   /// Calls the onLoad() method, which can be implemented on a Module.
   /// Executes the willLoad and didLoad event streams.
+  /// TODO: Do we want to support re-loading of child modules here?
```

Comment:
To me, it seems like a larger philosophical issue that will require some discussion across teams. It also seems like we can defer that discussion since it's gone this long without a resolution. I just wanted to make a note so we eventually deal with it.

If people would like to resolve it in this PR I'm game, we should probably talk on HC or GHO though.

@maxwellpeterson-wf @evanweible-wf thoughts?